### ISSUE TITLE
[Idea] Create a KongIngress instead of patching the existing Kong route

### ISSUE THREAD

**Issue by shanoor** [ollystephens/acme-kong-kube-helper#3]
Just a idea, wouldn't it better to create a matching KongIngress resource when a new Ingress is created? That would make it more permanent for Kong and there is no risk of losing the route customization because of outside event.

I'm not familiar enough with Go but I'll look into the code.

**Comment by ollystephens** [comment]
Yep, that's a good idea. Would need to ensure that the cluster role (`serviceaccount`) used had sufficient privileges, as would need to both create `KongIngress` objects, *and* patch `Ingress` objects. I think at the moment all it needs is permission to watch for changes. Also need to ensure that adding a `KongIngress` after the `Ingress` has been created actually triggers an update in the controller.
No promises though; this was a small piece of coding for what will hopefully be a short-term problem. I don't want to invest too much time in it.

**Comment by shanoor** [comment]
I did some digging: 
- the cluster role doesn't have enough permissions, it has ReadOnly on `kongingresses`
- adding `KongIngress` after is ok, it works as expected

I also hope the issue gets solved where it should be.
